front/src/common/useSearchUsers.ts

```diff
@@ -0,0 +1,64 @@
+import { useState, useCallback, useEffect } from 'react';
+
+import { useDebounce } from 'utils/helpers';
+
+import type { SearchResultItemUser } from './api/generatedEditoastApi';
+import type { Grant, SubjectType } from './api/mock/mockEditoastApi';
+import { osrdEditoastApi } from './api/osrdEditoastApi';
+
+export type UserWithPendingGrant = {
+  id: number;
+  type: SubjectType;
+  name: string;
+  grant?: Grant;
+}[];
+
+export default function useSearchUsers() {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [searchedUsers, setSearchedUsers] = useState<UserWithPendingGrant>([]);
+
+  const debouncedSearchTerm = useDebounce(searchTerm, 150);
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  const searchUser = useCallback(async (searchQuery: string) => {
+    try {
+      const results = (await postSearch({
+        searchPayload: {
+          object: 'user',
+          query: ['search', ['name'], searchQuery],
+        },
+        pageSize: 101,
+      }).unwrap()) as SearchResultItemUser[];
+
+      return results.map((result) => ({
+        id: result.id,
+        name: result.name,
+        type: 'user' as SubjectType,
```

Comment:
:warning: hum this cast is a bit dangerous...
```suggestion
        type: 'User' as SubjectType,
```